### PR TITLE
group: adds conditional move and select to group.

### PR DESCRIPTION
**Pull request by armfazh**
Adds conditional move and select operations for group elements and scalars.

### CODE REVIEW

File: group/group.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ type Element interface {
 	Copy() Element
 	IsIdentity() bool
 	IsEqual(Element) bool
+	CMov(int, Element) Element
```
Comment (bwesterb) [inline group/group.go:38]: Document.

File: group/group.go [hunk 2]
```diff
@@ -35,6 +35,8 @@ type Element interface {
...
+	CSelect(int, Element, Element) Element
```
Comment (bwesterb) [inline group/group.go:39]: Document.

File: group/group.go [hunk 3]
```diff
@@ -51,6 +53,8 @@ type Scalar interface {
 	Copy() Scalar
 	IsEqual(Scalar) bool
 	SetUint64(uint64)
+	CMov(int, Scalar) Scalar
```
Comment (bwesterb) [inline group/group.go:56]: Document

File: group/group.go [hunk 4]
```diff
@@ -51,6 +53,8 @@ type Scalar interface {
...
+	CSelect(int, Scalar, Scalar) Scalar
```
Comment (bwesterb) [inline group/group.go:57]: Document

File: group/ristretto255.go [hunk 5]
```diff
@@ -145,6 +147,17 @@ func (e *ristrettoElement) Copy() Element {
 	return &ristrettoElement{*new(r255.Point).Set(&e.p)}
 }
 
+func (e *ristrettoElement) CMov(v int, x Element) Element {
+	ee := (*edwards25519.ExtendedPoint)(&e.p)
```
Comment (bwesterb) [inline group/ristretto255.go:151]: This is frail. I've added Point.ConditionalSet in go-ristretto `v1.2.2`.

File: group/ristretto255.go [hunk 6]
```diff
@@ -197,6 +210,20 @@ func (s *ristrettoScalar) Copy() Scalar {
 	return &ristrettoScalar{*new(r255.Scalar).Set(&s.s)}
 }
 
+func (s *ristrettoScalar) CMov(v int, x Scalar) Scalar {
+	for i := range s.s {
+		s.s[i] = uint32(subtle.ConstantTimeSelect(v, int(x.(*ristrettoScalar).s[i]), int(s.s[i])))
+	}
+	return s
```
Comment (bwesterb) [inline group/ristretto255.go:217]: I've added Scalar.ConditionalSet in v1.2.2.

File: group/ristretto255.go [hunk 7]
```diff
@@ -197,6 +210,20 @@ func (s *ristrettoScalar) Copy() Scalar {
...
+}
+
+func (s *ristrettoScalar) CSelect(v int, x Scalar, y Scalar) Scalar {
+	for i := range s.s {
+		s.s[i] = uint32(subtle.ConstantTimeSelect(v, int(x.(*ristrettoScalar).s[i]), int(y.(*ristrettoScalar).s[i])))
```
Comment (bwesterb) [inline group/ristretto255.go:222]: I've added Scalar.ConditionalSet in v1.2.2.

**Review by bwesterb — CHANGES_REQUESTED** [review]
Looks good. I've added {Point,Scalar}.ConditionalSet to go-ristretto for you to use instead.

File: group/group.go (hunk 1 above)
Comment (bwesterb) [reply group/group.go:38]: In particular: what happens of the `int` isn't 0 or 1?

Comment (armfazh) [reply group/group.go:39]: All documentation was added in #355

**Review by bwesterb — APPROVED** [review]

**Pull request merged**